polling_stations/apps/dashboard/views.py

```diff
@@ -0,0 +1,173 @@
+import json
+import operator
+from functools import reduce
+
+from django.contrib.auth.mixins import UserPassesTestMixin
+from django.db import connection
+from django.db.models import Q
+from django.http import JsonResponse
+from django.urls import reverse
+from django.views import View
+from django.views.generic import TemplateView, DetailView, ListView
+
+from councils.models import Council
+from pollingstations.models import ResidentialAddress, PollingStation
+
+
+class DashboardMixin(UserPassesTestMixin):
+    def test_func(self):
+        return self.request.user.is_staff
+
+
+class IndexView(DashboardMixin, ListView):
+    queryset = Council.objects.select_related("dataquality")
+    template_name = "dashboard/council_list.html"
+
+
+class CouncilDetailView(DashboardMixin, DetailView):
+    queryset = Council.objects.all()
+    template_name = "dashboard/council_detail.html"
+
+    def get_context_data(self, object, **kwargs):
+        context = super().get_context_data(object=object, **kwargs)
+
+        with connection.cursor() as cursor:
+            cursor.execute(
+                """
+                SELECT postcode, st_maxdistance(multipoint, multipoint)::int AS maxdistance
+                FROM (
+                    SELECT postcode, st_transform(st_collect(location), 27700) AS multipoint
+                    FROM pollingstations_residentialaddress
+                    WHERE council_id = %s
+                    GROUP BY postcode
+                ) AS subquery
+                WHERE multipoint IS NOT NULL
+                ORDER BY maxdistance DESC
+                LIMIT 10;
```

Comment:
This query is useful (its essentially the non-visual form of plotting the lines as a materialized view and mapping them), but I find this is the form of this query I use least frequently because it ignores all the properties without UPRNs which are often the ones where errors are lurking. 

The variant where we base the 'home' point on the ONSPD centroid is good for catching errors on records where we've got a postcode but no UPRN, so would be useful to include that too. I find myself referring to that more often.

Also on these ones where we're listing addresses I'd generally look at way more than the first 10 records in SQL. I reckon we want the first 100 for this to be useful.